### ISSUE TITLE
sealed-secrets Controller Produces Recurring Ambiguous Events 

### ISSUE THREAD

**Issue by shin-go** [bitnami-labs/sealed-secrets#1284]
**Which component**:
controller

**Describe the bug**
After updating sealed-secrets to current stable (v0.23.0) it was observed that after all SealedSecret kind resources on Kubernetes cluster were unsealed, the following events were repeating ad neaseum:

```
WARNING: Empty API version & kind, filling it...                                                                                                        │
update suppressed, no changes in sealed secret spec of <namespace>/<sealed secret name>
```

**To Reproduce**

1. Encrypt and apply a SealedSecret on an older version of kubeseal+sealed-secrets. Around v0.20.0 should be old enough.
2. Update sealed-secrets to latest stable (v0.23.0)
3. Observe noted events recurring

**Expected behavior**
Maybe some documentation on what this is actually doing and why. There is no literature on why the update to SealedSecrets in place are suppressed - was there no change? Did it fail? The lack of clarity here makes it difficult to assess why the event happens continuously.

**Version of Kubernetes**:

AWS EKS 1.25.11

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.2", GitCommit:"7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647", GitTreeState:"clean", BuildDate:"2023-05-17T14:20:07Z", GoVersion:"go1.20.4", Compiler:"gc", Platform:"darwin/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"25+", GitVersion:"v1.25.11-eks-a5565ad", GitCommit:"df109513ba589adaee235b4e658476275d1fde31", GitTreeState:"clean", BuildDate:"2023-06-16T17:35:00Z", GoVersion:"go1.19.10", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**

**Comment by alvneiayu** [comment]
hi @shin-go 

Yes, totally right. This message is showed when the Sealed Secret applied is the same (no changes) but the message is a little bit short and no reason are showed. The best solution is to improve the message and share why is suppressed, IMO.

Thanks a lot for suggestion

Álvaro

**Comment by jnoordsij** [comment]
I ran into these warnings too, and especially the `WARNING: Empty API version & kind, filling it...` confuses me quite a bit and I don't really understand if there is any action required to properly fix the yaml manifests, or if it can be safely ignored...

**Comment by alvneiayu** [comment]
let me investigate a little bit @jnoordsij. It seems that it is showed when the kind or apiVersion are empty but I am seeing the error with the values in the YAML. Checking it but it is safe @jnoordsij, no worries.

**Comment by jnoordsij** [comment]
Yeah everything does seem to work fine, I ended up refreshing a bunch of my `SealedSecret`s and all of the corresponding `Secret`s were updated perfectly fine, just with this message showing up a lot of times. Just curious what's the cause or if there is anything wrong with my manifests that is fixable to prevent this.
